cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -306,16 +307,16 @@ data SomeWitnessSigningKey
                                (Api.SigningKey Api.GenesisDelegateExtendedKey)
   | AGenesisUTxOSigningKey     (Api.SigningKey Api.GenesisUTxOKey)
 
-readSigningKeyFile
-  :: SigningKeyFile
+readWitnessSigningData
+  :: WitnessSigningData
   -> ExceptT (Api.FileError SigningKeyDecodeError) IO SomeWitnessSigningKey
-readSigningKeyFile skFile =
+readWitnessSigningData (WitnessSigningData skFile mbByronAddr) =
```

Comment:
I should do better validation in this function. 

If an `Address Byron` is provided, an error should be returned if any other than a `AByronSigningKey` is deserialized from the `SigningKeyFile`.